src/basic/parse-util.c

```diff
@@ -70,26 +70,24 @@ int parse_pid(const char *s, pid_t* ret_pid) {
 }
 
 int parse_mode(const char *s, mode_t *ret) {
-        char *x;
-        long l;
+        unsigned m;
+        int r;
 
         assert(s);
-        assert(ret);
 
-        s += strspn(s, WHITESPACE);
-        if (s[0] == '-')
-                return -ERANGE;
```

Comment:
I don't think we should allow leading whitespace. Our parsers generally strip whitespace earlier, so if we get here with some leading whitespace, something strange is going on. But if the user somehow managed to feed us leading whitespace, we should fail here. But this behaviour didn't change with this PR, so let's leave that for later.